src/UI/SignTransactionsModals/SignWithExtensionModal/index.tsx

```diff
@@ -1,23 +1,33 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { faHourglass, faTimes } from '@fortawesome/free-solid-svg-icons';
-import { Modal } from 'react-bootstrap';
+import globalStyles from 'assets/sass/main.scss';
 import { SignModalPropsType } from 'types';
+import { useDappModal } from 'UI/DappModal';
+import { ModalContainer } from 'UI/ModalContainer/ModalContainer';
 import { PageState } from 'UI/PageState';
 import { getGeneratedClasses } from 'UI/utils';
 import { safeRedirect } from 'utils';
+import styles from './sing-with-extension-modal.scss';
 
 export const SignWithExtensionModal = ({
   handleClose,
   error,
   callbackRoute,
   transactions,
-  className = 'extension-modal'
+  className = styles.extensionModal
 }: SignModalPropsType) => {
   const classes = getGeneratedClasses(className, true, {
-    wrapper: 'modal-container extension',
-    icon: 'text-white',
-    closeBtn: 'btn btn-close-link mt-2'
+    wrapper: `${styles.modalContainer} ${styles.extension}`,
+    icon: globalStyles.textWhite,
+    closeBtn: `${globalStyles.btn} ${globalStyles.btnCloseLink} ${globalStyles.mt2}`
   });
+
+  const { handleShowModal } = useDappModal();
```

Comment:
I think we'll have a lot of similar cases. We can just pass in some props to useDappModal to make this more easy. for example, a prop for visibility here would make the useEffect below useless.

 useDappModal({openOnMount: true}), where the options object is optional, as well as all its options